eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -186,28 +184,46 @@ object Helpers {
   }
 
   /**
-   * The general rule is that we use remote_alias for our channel_update until the channel is publicly announced, and
+   * The general rule is that we use our local alias for our channel_update until the channel is publicly announced, and
```

Comment:
> The original comment is still accurate though? It described the spec logic of what scid should be used for communicating the channel_update to our direct peer and to the network.

Not really, because how we communicate the `channel_update` to our direct peer and how we communicate it to the network isn't the same? I find the comment much more logical now, we should build our `channel_update` using a real scid or a local alias, and the case where we send it directly to our peer and they support `scid_alias` is the exception, where we replace the scid by their alias (because it's really a technicality that they only know this alias so we must use it - as the comment on `channelUpdateForDirectPeer` explains).

> I would also implement the method differently and only rely on the ScidAlias feature bit.

Meh, that would make the code more complex because we'd need to explicitly handle the case where the `scid_alias` feature bit is set but they didn't send one, whereas now we can just implement all cases based on what data is available.

> Maybe things would be simpler if we renamed methods to:
>    scidForRelay
>    scidForNetwork

But these will have the same contents, we use the same scid in both cases?
There are only three distinct cases:

- the general case (network and relay)
- sending to a direct peer with `scid_alias` -> we must use their alias
- sending back a `TemporaryChannelFailure` -> we must use the scid provided in the onion when available

I think those three cases are quite clear the way it's implemented currently, isn't it?

> Maybe we should go even further and store the channel_update's contents and not the scid and signature. Rabbit-holey, I know.

You mean creating an `UnsignedChannelUpdate` that contains everything except `scid` and `signature`? The issue I see with that is that we'll end up re-signing more often than we need, which is a bit wasteful...the current code doesn't seem complex enough to really require that change?

Reply:
Updated with some of these changes in https://github.com/ACINQ/eclair/pull/2323/commits/c1c25ed74753210791bcca077421a31c49eebbc7